plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -0,0 +1,537 @@
+	fu_progress_set_steps(progress, chunks_isp->len);
+
+	for (guint i = 0; i < chunks_isp->len; i++) {
+		FuChunk *chk = g_ptr_array_index(chunks_isp, i);
+		g_autoptr(GByteArray) part_isp_data = g_byte_array_new();
+
+		fu_byte_array_append_uint8(part_isp_data,
+					   basic_data_size + fu_chunk_get_data_sz(chk));
+		fu_byte_array_append_uint8(part_isp_data, FU_ALGOLTEK_CMD_ISP);
+		fu_byte_array_append_uint16(part_isp_data, data_addr, G_BIG_ENDIAN);
+		g_byte_array_append(part_isp_data,
+				    fu_chunk_get_data(chk),
+				    fu_chunk_get_data_sz(chk));
+		fu_byte_array_append_uint8(part_isp_data,
+					   ~fu_sum8(part_isp_data->data, part_isp_data->len) + 1);
+
+		if (!algoltek_device_ctrl_transfer(self,
+						   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+						   FU_ALGOLTEK_CMD_ISP,
+						   0,
+						   0,
+						   part_isp_data,
+						   part_isp_data->data[0],
+						   error))
+			return FALSE;
+
+		data_addr += fu_chunk_get_data_sz(chk);
+		fu_progress_step_done(progress);
+	}
+
+	return TRUE;
+}
+
+static gboolean
+algoltek_device_bot(FuAlgoltekUsbDevice *self, int address, GError **error)
+{
+	g_autoptr(GByteArray) buf = g_byte_array_new();
+
+	fu_byte_array_set_size(buf, 11, 0x0);
+	buf->data[0] = 5;
+	buf->data[1] = FU_ALGOLTEK_CMD_BOT;
+	fu_memwrite_uint16(buf->data + 2, address, G_BIG_ENDIAN);
+	buf->data[10] = ~fu_sum8(buf->data, buf->len) + 1;
+
+	if (!algoltek_device_ctrl_transfer(self,
+					   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					   FU_ALGOLTEK_CMD_BOT,
+					   0,
+					   0,
+					   buf,
+					   buf->data[0],
+					   error))
+		return FALSE;
+
+	return TRUE;
+}
+
+static gboolean
+algoltek_device_ers(FuAlgoltekUsbDevice *self, GError **error)
+{
+	g_autoptr(GByteArray) buf = g_byte_array_new();
+
+	fu_byte_array_set_size(buf, 2, 0x0);
+
+	if (!algoltek_device_ctrl_transfer(self,
+					   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					   FU_ALGOLTEK_CMD_ERS,
+					   0,
+					   0,
+					   buf,
+					   buf->len,
+					   error))
+		return FALSE;
+	return TRUE;
+}
+
+static gboolean
+fu_algoltek_device_status_check(FuDevice *self, gpointer user_data, GError **error)
+{
+	g_autoptr(GByteArray) update_status_array = g_byte_array_new();
+	guint16 update_status;
+
+	update_status_array =
+	    algoltek_device_rdr(FU_ALGOLTEK_USB_DEVICE(self), AG_UPDATE_STATUS, error);
+	if (update_status_array == NULL)
+		return FALSE;
+
+	update_status = fu_algoltek_readout_value(update_status_array);
+
+	switch (update_status) {
+	case AG_UPDATE_PASS:
+		break;
+	case AG_UPDATE_FAIL:
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_INVALID_DATA,
+			    "Update procedure is failed.");
+		return FALSE;
+	default:
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+algoltek_device_wrf(FuAlgoltekUsbDevice *self,
+		    GBytes *blob_payload,
+		    FuProgress *progress,
+		    GError **error)
+{
+	g_autoptr(GByteArray) transfer_parameter = g_byte_array_new();
+	g_autoptr(GPtrArray) chunks_payload = NULL;
```

Comment:
can you put all the g_autoptr() bit *last* in the variable declaration block please? ie.. after `guint16 index`

Reply:
Removed the g_autoptr() bit after variable declaration block.